src/datasets/load.py

```diff
@@ -1772,31 +1773,30 @@ def dataset_module_factory(
                 else:
                     raise e
             if filename in [sibling.rfilename for sibling in dataset_info.siblings]:  # contains a dataset script
-                if config.USE_PARQUET_EXPORT:
+                fs = HfFileSystem(endpoint=config.HF_ENDPOINT, token=download_config.token)
+                with fs.open(f"datasets/{path}/{filename}", "r", revision=revision, encoding="utf-8") as f:
+                    can_load_config_from_parquet_export = (
+                        "DEFAULT_CONFIG_NAME" not in f.read() or _ignore_default_config
+                    )
+                if config.USE_PARQUET_EXPORT and can_load_config_from_parquet_export:
                     # If the parquet export is ready (parquet files + info available for the current sha), we can use it instead
+                    # This fails when the dataset has multiple configs and a default config and
+                    # the user didn't specify a configuration name (_ignore_default_config=False).
                     try:
                         return HubDatasetModuleFactoryWithParquetExport(
                             path, download_config=download_config, revision=dataset_info.sha
                         ).get_module()
                     except _datasets_server.DatasetsServerError:
                         pass
                 # Otherwise we must use the dataset script if the user trusts it
-                trust_remote_code = resolve_trust_remote_code(trust_remote_code, path)
```

Comment:
I guess this is no longer necessary?

Reply:
It's already called inside the HubDatasetModuleFactoryWithScript - no need to have it here